src/dataframe/io.jl

```diff
@@ -561,11 +561,9 @@ function builddf(rows::Integer,
                     continue
                 else
                     msgio = IOBuffer()
-                    @printf(msgio,
-                            "Failed to parse '%s' using type '%s'",
-                            bytestring(p.bytes[left:right]),
-                            o.eltypes[j])
-                    error(bytestring(msgio))
+                    print(msgio, "Failed to parse '", p.bytes[left:right],
+                        "' using type ", o.eltypes[j])
```

Comment:
But then also raise the dependency to the latest Compat version. You could also simplify the code by using `@sprintf` instead of `@printf`, and pass the result to `error` without using a buffer.